crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -735,6 +735,54 @@ export.input_note_get_assets_info
     # => [ASSETS_COMMITMENT, num_assets, pad(11)]
 end
 
+#! Returns the recipient commitment of the input note with the specified index.
+#!
+#! Inputs:  [note_index, pad(15)]
+#! Outputs: [RECIPIENT_COMMITMENT, pad(12)]
+#!
+#! Where:
+#! - note_index is the index of the input note whose recipient commitment should be returned.
+#! - RECIPIENT_COMMITMENT is the commitment to the input note's script, inputs, the serial number.
+#!
+#! Invocation: dynexec
+export.input_note_get_recipient
+    # get the note data pointer by the provided index
+    exec.memory::get_input_note_ptr
+    # => [note_ptr, pad(15)]
+
+    # compute the recipient commitment
+    exec.input_note::compute_recipient_commitment
+    # => [RECIPIENT_COMMITMENT, pad(15)]
```

Comment:
This would re-compute the recipient on every invocation, right? If so, why don't we compute the recipient in the prologue, save it to memory, and then here just read it from memory?

Reply:
Looks like previously we used the recipient hash only for the note ID computation, we didn't use it directly, for this reason we didn't save it in memory. But I agree that it makes sense to start doing so, I'll add the recipient to the note data segment.